controllers/users.js

```diff
@@ -466,6 +468,74 @@ const filterUsers = async (req, res) => {
   }
 };
 
+const DISCORD_BASE_URL = config.get("services.discordBot.baseUrl");
+
+// wroker queue created with Bull
+const syncQueue = new Queue("syncQueue", {
+  redis: {
+    port: REDIS_PORT,
+    host: REDIS_URI,
+  },
+});
+
+/*
+ * Sync user data with the data present in our discord server
+ *
+ */
+
+const syncInDiscordRole = async (req, res) => {
+  try {
+    // create a authToken to verify on discord bot
+    const authToken = jwt.sign({}, config.get("rdsServerlessBot.rdsServerLessPrivateKey"), {
+      algorithm: "RS256",
+      expiresIn: config.get("rdsServerlessBot.ttl"),
+    });
+
+    // fetch the data of member in discord server
+    const response = await fetch(`${DISCORD_BASE_URL}/discord-members`, {
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${authToken}`,
+      },
+    });
+    const discordMembers = await response.json();
+
+    // get the data of the users from our database
+    const allUsers = await userQuery.getAllUsers();
+
+    allUsers.forEach((doc) => {
+      const user = { id: doc.id, ...doc.data() };
+
+      if (user.roles.archived) {
+        // Update: If the user is archived change the role in_discord: false
+        user.roles = { ...user.roles, in_discord: false };
+        syncQueue.add({ user });
+
+        // If the user has verified them self and has discordId in the database
+      } else if (user.discordId) {
+        // find that users data from the discord members data we have
+        const discordUserData = discordMembers.find((item) => item.user.id === user.discordId);
+
+        // if data exists that means the user is present in our discord derver
+        if (discordUserData) {
+          // Update: the user role in_dicord: true and the joined_discord date
+          user.roles = { ...user.roles, in_discord: true };
+          user.joined_discord = discordUserData.joined_at;
+          syncQueue.add({ user });
+        } else {
+          user.roles = { ...user.roles, in_discord: false };
+          syncQueue.add({ user });
+        }
+      }
+    });
+
+    return res.json({ message: "Synced with discord members." });
+  } catch (error) {
+    logger.error(`Error while fetching all users: ${error}`);
+    return res.boom.serverUnavailable("Something went wrong please contact admin");
```

Comment:
the error message is used at multiple places. would be good if we can import from a constant and then use it.

Reply:
done